apps/condo/domains/ticket/utils/serverSchema/resolveHelpers.js

```diff
@@ -2,7 +2,33 @@ const get = require('lodash/get')
 const { getSectionAndFloorByUnitName } = require('@condo/domains/ticket/utils/unit')
 const { Property } = require('@condo/domains/property/utils/serverSchema')
 const { Contact } = require('@condo/domains/contact/utils/serverSchema')
-const { TICKET_ORDER_BY_STATUS, STATUS_IDS } = require('@condo/domains/ticket/constants/statusTransitions')
+const { TICKET_ORDER_BY_STATUS, STATUS_IDS, TERMINAL_TICKET_STATUS_IDS } = require('@condo/domains/ticket/constants/statusTransitions')
+const { COMPLETED_STATUS_TYPE, NEW_OR_REOPENED_STATUS_TYPE } = require('@condo/domains/ticket/constants')
+const { TicketStatus } = require('@condo/domains/ticket/utils/serverSchema')
+
+async function updateResolvedDataByResolvedStatus (context, existingItem, resolvedData) {
+    const existedStatusId = get(existingItem, 'status')
+    const resolvedStatusId = get(resolvedData, 'status')
+
+    if (resolvedStatusId) {
+        addOrderToTicket(resolvedData, resolvedStatusId)
+
+        if (TERMINAL_TICKET_STATUS_IDS.includes(resolvedStatusId)) {
+            resolvedData.deadline = null
+        }
+
+        if (existedStatusId) {
+            const existedStatus = await TicketStatus.getOne(context, { id: existedStatusId })
+            const resolvedStatus = await TicketStatus.getOne(context, { id: resolvedStatusId })
+
+            if (get(existedStatus, 'type') === COMPLETED_STATUS_TYPE && get(resolvedStatus, 'type') === NEW_OR_REOPENED_STATUS_TYPE) {
+                const existedStatusReopenedCounter = existingItem['statusReopenedCounter']
+                resolvedData['statusReopenedCounter'] = existedStatusReopenedCounter + 1
+                resolvedData['executor'] = null
+            }
+        }
+    }
+}
```

Comment:
it's too complex. You have here at least three different things:
 1) update ticket order
 2) calculate reopened counter
 3) clean deadline and clean executor (is it really needed?)

It's better to split it to three different functions